client/src/controllers/BlockController.ts

```diff
@@ -0,0 +1,71 @@
+import { Controller } from '@hotwired/stimulus';
+
+declare global {
+  interface Window {
+    initBlockWidget?: (id: string) => void; // Declare the initBlockWidget function as optional on the Window interface
```

Comment:
```suggestion
    initBlockWidget?: (id: string) => void;
```

No need for this extra comment.